src/Sentry/Internal/MainExceptionProcessor.cs

```diff
@@ -49,7 +56,30 @@ public void Process(Exception exception, SentryEvent sentryEvent)
 
                 foreach (var key in sentryException.Data.Keys)
                 {
-                    sentryEvent.SetExtra($"Exception[{i}][{key}]", sentryException.Data[key]);
+                    if (key.StartsWith("sentry", StringComparison.OrdinalIgnoreCase))
+                    {
+                        if (key.StartsWith(ExceptionDataTagKey, StringComparison.OrdinalIgnoreCase))
+                        {
+                            var data = sentryException.Data[key] as string ?? "<invalid>";
```

Comment:
I rather we drop the value (internal logging as error) since we expect this to be `string` (and nothing else, not calling `ToString()` either) since the API to get this key is `AddSentryTag(string,string)`